src/ui/public/courier/data_source/build_query/from_kuery.js

```diff
@@ -0,0 +1,52 @@
+import _ from 'lodash';
+import { fromKueryExpression } from '../../../kuery';
+
+function astToQuery(ast) {
+  if (Array.isArray(ast)) {
+    const query = {
+      must: [],
+      should: [],
+      must_not: [],
+    };
+
+    ast.forEach((astClause) => {
+      const clause = astToQuery(astClause);
+      if (astClause.negate) {
+        query.must_not.push(clause);
+      }
+      else {
+        query.should.push(clause);
+      }
+    });
+
+    return query;
+  }
+  else {
+    if (ast.type === 'match') {
+      return {
+        match: {
+          [ast.params.field]: ast.params.value
+        }
+      };
+    }
+    else if (ast.type === 'range') {
+      return {
+        range: {
+          [ast.params.field]: {
+            gt: ast.params.gt,
+            lt: ast.params.lt
+          }
```

Comment:
I think we can make it work, I definitely want to investigate it. But TBH I'm thinking pluggability is like, phase 3 at the earliest. Post 6.0.